test/smoketest_artman.py

```diff
@@ -16,98 +16,117 @@
 
 """Artman smoke tests.
 
-It generates both grpc and gapic client libraries for Google APIs in googleapis
+It generates GAPIC client libraries for Google APIs in googleapis
 repo (except those being blacklisted), and fails if any generation fails.
-The test currently assumes that artman and googleapis folders are under the
-same parent folder.
 """
 
 import argparse
-import fnmatch
+import glob
+import json
+import logging
 import os
-import re
 import subprocess
 import sys
+import yaml
 
-ARTMAN_CONFIG_BLACKLIST = [
-    'artman_spanner_admin_instance.yaml',
-    'artman_spanner_admin_database.yaml',
-    'artman_bigtable_admin.yaml'
-]
-
-SUPPORTED_LANGS = ['python', 'java', 'ruby', 'nodejs', 'php', 'go', 'csharp']
+from google.protobuf import json_format
 
+from artman.config.proto import config_pb2
 
-def run_smoke_test(apis, input_dir):
-    input_dir = os.path.abspath(input_dir)
-    artman_config_dirs = [
-        os.path.join(input_dir, 'gapic', 'api'),
-    ]
-
-    artman_config_whitelist = []
-    if apis:
-        for api in apis.split(','):
-            artman_config_whitelist.append('artman_%s.yaml' % api)
+CONFIG_BLACKLIST = [
```

Comment:
I'm a little concerned about the maintainability of such a blacklist. Will we remember to update once these should succeed? It would be good if each line linked to an issue describing what is failing and what needs to be done to make it succeed.

Reply:
I share your concern. I don't plan to file GitHub issues as there won't be assignees except myself. I do have a plan to address each of them. Another option is to run generation for those in the blacklist but not count them as errors but warnings if the execution fails.

Reply:
Makes sense. Could you either:

- file an uber-issue to remove this blacklist once each case is addressed
- run these generations, and only warn, not fail

so that we'll have some tracking?